mlflow/johnsnowlabs.py

```diff
@@ -0,0 +1,874 @@
+                    os.environ.update({k: str(v) for k, v in json.load(open(license_path)).items() if v is not None})
+                    os.environ['JSL_NLP_LICENSE'] = json.load(open(license_path))['HC_LICENSE']
+            print("STARTING NEW  SESSION WITH JARS: ", jar_paths)
+            spark = nlp.start(nlp=False, spark_nlp=False, jar_paths=jar_paths, json_license_path=license_path,
+                              create_jsl_home_if_missing=False, spark_conf=spark_conf)
+        else:
+            spark = nlp.start()
+    return spark
+
+
+def create_sparksession_for_jsl_uri(**kwargs):
+    # if using johnsnowlabs:/ uri, we need to start a sparksession
+    # Support johnsnowlabs nlp.start() API to fetch a session
+    nlp.start(**kwargs)
+
+
+def fetch_deps_from_path(local_model_path):
+    if _JOHNSNOWLABS_MODEL_PATH_SUB not in local_model_path:
+        local_model_path = os.path.join(local_model_path, _JOHNSNOWLABS_MODEL_PATH_SUB, 'jars.jsl', )
+    else:
+        local_model_path = os.path.join(local_model_path, 'jars.jsl', )
+
+    jar_paths = [os.path.join(local_model_path, file) for file in os.listdir(local_model_path) if '.jar' in file]
+    license_path = [os.path.join(local_model_path, file) for file in os.listdir(local_model_path) if '.json' in file]
+    license_path = license_path[0] if license_path else None
+    return jar_paths, license_path
+
+
+def _unpack_and_save_model(spark_model, dst):
+    from pyspark.ml import PipelineModel
+    if isinstance(spark_model, _PyFuncModelWrapper):
+        spark_model = spark_model.spark_model
+    if isinstance(spark_model, PipelineModel):
+        spark_model.write().overwrite().save(dst)
+    else:
+        # nlu pipe
+        spark_model.predict('Init')
+        try:
+            spark_model.vanilla_transformer_pipe.write().overwrite().save(dst)
+        except:
+            # for mlflowdbfs_path we cannot use overwrite, gives
+            # org.apache.hadoop.fs.UnsupportedFileSystemException: No FileSystem for scheme "mlflowdbfs"
+            spark_model.save(dst)
+
+
+class _PyFuncModelWrapper:
+    """
+    Wrapper around NLUPipeline providing interface for scoring pandas DataFrame.
+    """
+
+    def __init__(self, spark_model, spark=None, ):
+        # we have this ternary, so we support _PyFuncModelWrapper(nlu_ref)
+        self.spark = spark if spark else get_or_create_sparksession()
+        self.spark_model = spark_model
+
+    def predict(self, text, output_level=''):
+        """
+        Generate predictions given input data in a pandas DataFrame.
+
+        :param output_level:
+        :param text: pandas DataFrame containing input data.
+        :return: List with model predictions.
+        """
+        return self.spark_model.predict(text, output_level=output_level).reset_index().to_json()
+
+
+@autologging_integration(FLAVOR_NAME)
+def autolog(disable=False, silent=False):  # pylint: disable=unused-argument
```

Comment:
I don't think we need autolog. Can we remove it?

Reply:
ok makes sense, we already have this in `mlflow.spark` and it adds no use here right? thanks